pandas/core/groupby/generic.py

```diff
@@ -442,9 +442,16 @@ def transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
         )
 
     def _cython_transform(
-        self, how: str, numeric_only: bool = True, axis: int = 0, **kwargs
+        self, how: str, numeric_only: bool = False, axis: int = 0, **kwargs
     ):
         assert axis == 0  # handled by caller
+        if numeric_only:
+            warnings.warn(
+                f"Passing `numeric_only=True` to {type(self).__name__}.{how} is "
+                f"deprecated and will raise in a future version of pandas.",
+                category=FutureWarning,
+                stacklevel=find_stack_level(),
+            )
```

Comment:
With the exception of `rank`, all Series ops fail when passing numeric_only=True (even when the Series is numeric) with a NotImplementedError. However most SeriesGroupBy ops allow it. SerisGroupBy ops will be successful when they can handle object dtype or the dtype is numeric. They will raise an error attempting to perform the op when dtype cannot be operated on.

I'm thinking we should make SeriesGroupBy consistent with Series by raising whenever numreic_only is truthy.

Reply:
When I first adding the deprecation, I thought other SeriesGroupBy ops would always raise when passing numreic_only=True. Seeing now that's not the case, going to siphon this off into a followup.